modules/compute/src/integrationTest/java/org/apache/ignite/internal/compute/threading/ItComputeApiThreadingTest.java

```diff
@@ -222,4 +281,40 @@ CompletableFuture<?> executeOn(JobExecution<?> execution) {
             return action.apply((JobExecution<Object>) execution);
         }
     }
+
+    private enum ComputeMapReduceOperation {
+        SUBMIT_MAP_REDUCE(compute -> compute
+                .submitMapReduce(TaskDescriptor.builder(NoOpMapReduceTask.class).build(), null)
+        );
+
+        private final Function<IgniteCompute, TaskExecution<?>> action;
+
+        ComputeMapReduceOperation(Function<IgniteCompute, TaskExecution<?>> action) {
+            this.action = action;
+        }
+
+        TaskExecution<?> executeOn(IgniteCompute compute) {
+            return action.apply(compute);
+        }
+    }
+
+    private enum TaskExecutionAsyncOperation {
+        STATES_ASYNC(execution -> execution.statesAsync()),
```

Comment:
I wonder why you didn't use method references, they seem fitting. Is this a stylistic choice?

Reply:
Some of this lambdas do their calls without parameters and can be converted to method references, but some supply parameters and hence cannot. Current approach when everything looks similarly seems to be more consistent to me.